lib/resources/r4/document_reference.yaml

```diff
@@ -2,12 +2,76 @@
 name: DocumentReference
 field_name_base_url: https://www.hl7.org/fhir/documentreference-definitions.html#DocumentReference
 fields:
+- name: resourceType
+  required: 'Yes'
+  type: string
+  description: The type of the FHIR resource.
+  example: |
+    {
+      "resourceType": "DocumentReference"
+    }
+  note: resourceType must be DocumentReference.
+  url: http://hl7.org/fhir/r4/documentreference.html
+
+- name: status
+  required: 'Yes'
+  type: code
+  action:
+    - create
+  description: current
+  note: The status of this document reference.
+  url: http://hl7.org/fhir/r4/documentreference-definitions.html#DocumentReference.status
+  example: |
+    {
+      "status": "current"
+    }
+  binding:
+    description: The status of the document reference.
+    terminology:
+      - display: DocumentReference status
+        system: http://hl7.org/fhir/document-reference-status
+        info_link: https://www.hl7.org/fhir/valueset-document-reference-status.html
+
+- name: docStatus
+  required: 'Yes'
+  type: code
+  action:
+    - create
+  description: final | amended
+  note: The status of the underlying document.
+  url: http://hl7.org/fhir/r4/documentreference-definitions.html#DocumentReference.docStatus
+  example: |
+    {
+      "docStatus": "final"
+    }
+  binding:
+    description: Status of the underlying document.
+    terminology:
+      - display: DocumentReference docStatus
+        system: http://hl7.org/fhir/r4/codesystem-composition-status.html
+        info_link: http://hl7.org/fhir/r4/valueset-composition-status.html
+
 - name: type
   required: 'No'
   type: CodeableConcept
-  description: Precise type of clinical document
+  action:
+    - create
+  description: Specifies the particular kind of document referenced.
+  url: http://hl7.org/fhir/r4/documentreference-definitions.html#DocumentReference.type
+  example: |
+    {
+      "type": {
+        "coding": [
+          {
+            "system": "http://loinc.org",
+            "code": "83805-2",
+            "userSelected": false
+          }
+        ],
+      }
+    }
   binding:
-    description: Precise type of clinical document
+    description: Specifies the particular kind of document referenced.
```

Comment:
For type, we'll need to do something similar to dstu2, with a note stating we accept both proprietary and standard codes.

Reply:
You can find the changes here: https://github.com/cerner/fhir.cerner.com/pull/446/commits/c11917e277b1bde8d54f91f63e4134060da17f1e